src/data/markdown/docs/02 javascript api/11 jslib/functional/01 Intruduction.md

```diff
@@ -0,0 +1,92 @@
+---
+title: 'Getting started with functional.js'
+description: 'Introduction to functional testing in k6.'
+---
+
+The design of the `functional` library was inspired by Ava, Jest and Jasmine. If you already know one of these frameworks, using this library should be very simple. 
+
+The `functional` library is built on top of the built-in `check` and `group` k6 APIs.
+
+
+## Installation
+There's nothing to install. This library is hosted on jslib and can be imported in the k6 script directly.
+
+<CodeGroup labels={[]}>
+
+```javascript
+import { test } from 'https://jslib.k6.io/functional/0.0.2/index.js';
+```
+
+</CodeGroup>
+
+Alternatively, you can use a copy of this file stored locally.
+
+## Simple example
+
+Let's get started by writing a test for a hypothetical http API that should return a JSON array of objects. 
+
+First, create a `mytest.js` k6 script file.
+
+
+<CodeGroup labels={[]}>
+
+```javascript
+import { test } from 'https://jslib.k6.io/functional/0.0.2/index.js';
+import http from 'k6/http';
+
+export default function testSuite() {
+  test('Basic API test', (t) => {
+    let response = http.get("https://test-api.k6.io/public/crocodiles")
+
+    t.expect(response.status).as("API status code").toEqual(200);
```

Comment:
Why is `expect` a member of `t` rather than a separate import?

Reply:
expect() raises a very specific Exception when the condition is false. The `test()` catches that exception and processes the result. If `expect` is called outside of `test()`, the exception will abort iteration and produce a very confusing stack trace. This is why `expect` is not available outside of `test`. 
Ava does it the same way, so that's what I opted for :-)